tests/tests.go

```diff
@@ -289,6 +297,30 @@ func ChangeManagerLeadership() {
 	}, 2*time.Minute, 3*time.Second).Should(Equal(corev1.ConditionTrue), "should have a leader manager pod with ready condition")
 }
 
+func waitManagerDeploymentReady() {
+	By("Waiting for manager deployment to be ready")
+	Eventually(func() (corev1.ConditionStatus, error) {
+		managerDeployment, err := testClient.KubeClient.AppsV1().Deployments(managerNamespace).Get(context.TODO(), names.MANAGER_DEPLOYMENT, metav1.GetOptions{})
+		if err != nil {
+			return corev1.ConditionUnknown, err
+		}
+		pods, err := getKubemacpoolPods()
+		if err != nil {
+			return corev1.ConditionUnknown, err
+		}
+		if int32(len(pods.Items)) != *managerDeployment.Spec.Replicas {
+			return corev1.ConditionUnknown, nil
+		}
+		for _, condition := range managerDeployment.Status.Conditions {
+			if condition.Type == v1.DeploymentAvailable {
+				return condition.Status, nil
+			}
+		}
+		return corev1.ConditionUnknown, nil
+	}, 2*time.Minute, 3*time.Second).Should(Equal(corev1.ConditionTrue), "should have available deployment after chaing numbe of replicas")
```

Comment:
I think the `Eventually` error message should not mention replicas change in `waitManagerDeploymentReady` function scope, instead this function should return an error then handle it in the caller function.